src/petals/server/backend.py

```diff
@@ -48,25 +48,28 @@ def __init__(self, *args, memory_cache: MemoryCache, backend_dtype: torch.dtype,
             self.kwargs_schema,
         )
 
-    def inference_step(self, cache_metadata: torch.IntTensor, *inputs: torch.Tensor) -> Tuple[torch.Tensor, ...]:
```

Comment:
Fixed bug: `PrioritizedTaskPool` calculates the "task size" using the first 2 dims of the 1st arg to this function. The 2 dims of cache_metadata were `(2, batch_size)` and didn't include `prefix_length`, so effectively the task size was not enforced for inference (a user could send an inference with prefix_length = 2048 and batch_size = 2048!).